### PR TITLE
fix: implement icon in backend and show icon in table if available

### PR DESCRIPTION
**Pull request by dweber019**
## Hey, I just made a Pull Request!

Fixes https://github.com/backstage/backstage/issues/26255

CleanShot 2024-09-03 at 00 02 01@2x

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
Nice, LGTM!

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx [hunk 1]
```diff
@@ -185,6 +187,15 @@ export const NotificationsTable = ({
   const compactColumns = React.useMemo((): TableColumn<Notification>[] => {
     const showToolbar = notifications.length > 0;
 
+    const NotificationIcon = (props: { notification: Notification }) => {
+      if (props.notification.payload.icon) {
+        const Icon =
+          app.getSystemIcon(props.notification.payload.icon) ?? SvgIcon;
+        return <Icon />;
+      }
+      return <SeverityIcon severity={props.notification.payload?.severity} />;
```
Comment (mareklibra) [inline plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx:196]: unnecessary optional chaining `?` operator

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx [hunk 2]
```diff
@@ -185,6 +187,15 @@ export const NotificationsTable = ({
   const compactColumns = React.useMemo((): TableColumn<Notification>[] => {
     const showToolbar = notifications.length > 0;
 
+    const NotificationIcon = (props: { notification: Notification }) => {
```
Comment (mareklibra) [inline plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx:190]: Can we move definition of this component out of `NotificationsTable`? Up in this file.
The `useApp` will need to be called again in it.

**Review by mareklibra — COMMENTED** [review]
Thank you for your PR. I have just minor nitpicks, otherwise lgtm.

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx (hunk 1 above)
Comment (dweber019) [reply plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx:196]: removed

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx (hunk 2 above)
Comment (dweber019) [reply plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx:190]: extracted notification component to separate file.

**Review by mareklibra — APPROVED** [review]

**Review by jhaals — APPROVED** [review]
Nice, thanks all! 👍

**Pull request merged**